### PR TITLE
added user update profession functionality

### PR DESCRIPTION
**Pull request by gideonnnalue**
Desc: Added user update functionality

### CODE REVIEW

File: src/components/profile/actions.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+export const updateUserProfile = (data, close) => ({
+	type: actionTypes.UPDATE_USER,
+	payload: { data, close },
+});
+
+export const updateUserProfileSuccess = data => ({
+	type: actionTypes.UPDATE_USER_SUCCESS,
+	payload: data,
+});
+
```
Comment (EngrEric) [inline src/components/profile/actions.js]: try to practice DRY-do not repeat yourself
you can simply destructure `const {UPDATE_USER_SUCCESS, UPDATE_USER, ...} = actionTypes;`

**Comment by gideonnnalue** [comment]
> @gideonnnalue i do not know why you are not requesting for mentors review but your team members only

I wanted them to review something, I wasnt done yet

**Comment by Way2nnadi** [comment]
@gideonnnalue Why does your PR have over 250 lines of changes?

File: src/components/profile/reducer.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+} from './actionTypes';
+
+const INITIAL_STATE = {
+	profile: null,
+	loading: false,
+	error: null,
+};
+
+export default (state = INITIAL_STATE, action) => {
+	switch (action.type) {
```
Comment (EngrEric) [inline src/components/profile/reducer.js]: `const { type, payload } = action`
`switch (type)`
. . .

File: src/components/profile/sagas.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+	updateUserProfileSuccess,
+} from './actions';
+import { notificationsSettings } from './constants';
+import { getCurrentUserProfile, updateCurrentUserProfile } from './services';
+
+function* getCurrentUser() {
+	try {
+		const res = yield call(getCurrentUserProfile);
+		if (res.data.status === 'success') {
+			yield put(getUserProfileSuccess(res.data.data));
```
Comment (EngrEric) [inline src/components/profile/sagas.js]: `const { data, status } = res.data`
`if (status === 'success') {`
. . .

**Pull request merged**